packages/ensnode-schema/src/schemas/registrars.schema.ts

```diff
@@ -0,0 +1,358 @@
+     * Definition of "incremental duration" is
+     * the incremental increase in the lifespan of the registration for
+     * `node` that was active as of `blockTimestamp`.
+     *
+     * Please consider the following situation:
+     *
+     * A registration of direct subname of .eth name is scheduled to expire on
+     * Jan 1, midnight UTC. It is currently 30 days after this expiration time.
+     * Therefore, there are currently another 60 days of grace period remaining
+     * for this name. Anyone can now make a renewal of this name.
+     *
+     * There are two possible scenarios when a renewal is made:
+     *
+     * 1) If a renewal is made for 10 days incremental duration,
+     *    this name remains in an "expired" state, but it now
+     *    has another 70 days of grace period remaining.
+     *
+     * 2) If a renewal is made for 50 days incremental duration,
+     *    this name is no longer "expired" and is active, but it now
+     *    expires in 20 days.
+     *
+     * After the latest registration of a direct subname becomes expired by
+     * more than the grace period, it can no longer be renewed by anyone.
+     * It must first be registered again, starting a new registration lifecycle of
+     * expiry / grace period / etc.
+     *
+     * Guaranteed to be a non-negative bigint value.
+     */
+    incrementalDuration: t.bigint().notNull(),
+
+    /**
+     * Base cost of the Logical Registrar Action.
+     *
+     * Guaranteed to be:
+     * 1) null if and only if `total` is null.
+     * 2) Otherwise, a non-negative bigint value for registrations.
+     */
+    baseCost: t.bigint(),
+
+    /**
+     * Premium of the Logical Registrar Action.
+     *
+     * Guaranteed to be:
+     * 1) null if and only if `total` is null.
+     * 2) Otherwise, zero when `type` is `renewal`.
+     * 3) Otherwise, a non-negative bigint value `type` is `registration`.
+     */
+    premium: t.bigint(),
+
+    /**
+     * Total cost of performing Logical Registrar Action.
+     *
+     * Guaranteed to be:
+     * 1) null if and only if both `baseCost` and `premium` are null.
+     * 2) Otherwise, a non-negative bigint value, equal to the sum of
+     *    `baseCost` and `premium`.
+     */
+    total: t.bigint(),
+
+    /**
+     * Account that initiated the Logical Registrar Action and
+     * is paying the `total` cost.
+     */
+    registrant: t.hex().notNull(),
+
+    /**
+     * Encoded Referrer
+     *
+     * Represents the "raw" 32-byte "referrer" value emitted onchain in
+     * association with the registrar action.
+     *
+     * If a registrar / registrar controller doesn't support the concept of
+     * referrers then this field is set to null.
+     *
+     * Guaranteed to be:
+     * 1) null if a registrar / registrar controller doesn't support
+     *    the concept of referrers.
+     * 2) Otherwise, a hex string representation of 32-bytes.
+     */
+    encodedReferrer: t.hex(),
+
+    /**
+     * Decoded referrer
```

Comment:
```suggestion
     * Decoded referrer according to the subjective interpretation of `encodedReferrer` defined for
     * ENS Holiday Awards.
```

Please note the importance of this distinction. It's critical!!